### ISSUE TITLE
VS Manage NuGet Packages support for CentralPackageVersions

### ISSUE THREAD

**Issue by japj** [microsoft/MSBuildSdks#62]
When I use the VS buildin Manage NuGet Packages to update package versions, it adds the Version attribute back to the PackageReference instead of modifying the Packages.props file.

This breaks the integration

**Comment by jeffkl** [comment]
The NuGet team is actively working on this at https://github.com/NuGet/Home/issues/6764

**Comment by stevenbrix** [comment]
@jefflk, will there be any value in this MSBuild SDK once the Nuget team adds this support? It looks like they are building the exact same thing (which is good, this should be a feature of Nuget and not something the MSBuild team had to add on)

**Comment by jeffkl** [comment]
We will most likely deprecate `Microsoft.Build.CentralPackageVersions` once NuGet has built in support for managing package versions centrally.  I have been working with the NuGet team on their implementation and I'll probably work on some tooling to help people migrate.

**Comment by jmecosta** [comment]
Good to hear! Is there any ETA for it?

**Comment by jeffkl** [comment]
Yes it should be available in Visual Studio 2019 16.6 which is due out in May.  Preview versions will have it sooner.  It's currently [in a feature branch](https://github.com/NuGet/NuGet.Client/tree/dev-feature-cpvm) waiting on [an MSBuild feature](https://github.com/microsoft/msbuild/pull/4975).

**Comment by jmecosta** [comment]
Alright, thats super good news. Thanks for the update

**Comment by christianbumann** [comment]
> 
> 
> Yes it should be available in Visual Studio 2019 16.6 which is due out in May. Preview versions will have it sooner. It's currently [in a feature branch](https://github.com/NuGet/NuGet.Client/tree/dev-feature-cpvm) waiting on [an MSBuild feature](https://github.com/microsoft/msbuild/pull/4975).

@jeffkl This isn't working for me in Visual Studio Version 16.7. Is there anything I need to consider to get this working properly? I am also not able to find anything in the Visual Studio 16.6 release notes (bug fixes)

**Comment by jeffkl** [comment]
@christianbumann Are you having problems with our Microsoft.Build.CentralPackageVersions or the NuGet Central Package Version Management?  For support on the NuGet solution to central package version management, you'll need to look over at https://github.com/NuGet/Home/issues?q=is%3Aopen+is%3Aissue+label%3AArea%3ARestoreCPVM

**Comment by christianbumann** [comment]
> 
> 
> @christianbumann Are you having problems with our Microsoft.Build.CentralPackageVersions or the NuGet Central Package Version Management? For support on the NuGet solution to central package version management, you'll need to look over at https://github.com/NuGet/Home/issues?q=is%3Aopen+is%3Aissue+label%3AArea%3ARestoreCPVM

@jeffkl Its the `Microsoft.Build.CentralPackageVersions`. I just tested it with a simple project with following files

packages file inside a subfolder. e.g. src\test\test.csproj
``` 
<Project Sdk="Microsoft.NET.Sdk.Web">
  <Sdk Name="Microsoft.Build.CentralPackageVersions" Version="2.0.79" />
  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <OutputType>library</OutputType>
    <PackageId>test</PackageId>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging" />
  </ItemGroup>
</Project>
```

packages.props inside the src folder. e.g. src\packages.props

```
<?xml version="1.0" encoding="utf-8"?>
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <ItemGroup>
      <PackageReference Update="Microsoft.Extensions.Logging" Version="5.0.0" />
  </ItemGroup>
</Project>
```

Updating this package with the Visual Studio package manager will update the PackageReference inside the *.csproj instead of the packages.props, and new packages are added to the *.csproj instead of the packages.props.

I'am using am not on VS 16.8.3, not 16.7.3 as mentioned above.

**Comment by jeffkl** [comment]
@christianbumann sorry for the confusion, `Microsoft.Build.CentralPackageVersions` is not integrated with Visual Studio and probably never will be.  The NuGet team has been working on an integrated solution which shipped in a preview form in 16.6.  Their solution will eventually supersede `Microsoft.Build.CentralPackageVersions`.

For the NuGet implementation, please refer to https://github.com/NuGet/Home/issues/6764

**Comment by japj** [comment]
@jeffkl with the provided information I think it makes sense that I close this issue in the MSBuildSdk repo, since the MS Roadmap is to have the new NuGet/VS integration handle it (out of scope for this repo)
